src/components/note/note.test.tsx

```diff
@@ -0,0 +1,162 @@
+      noteContent={EditorState.createEmpty()}
+      title="Title"
+    />
+  );
+
+  expect(screen.getByRole("banner")).toHaveTextContent("Title");
+});
+
+test("should render with provided `name` prop", () => {
+  render(
+    <Note
+      createdDate="23 May 2020, 12:08 PM"
+      noteContent={EditorState.createEmpty()}
+      name="Carbon"
+    />
+  );
+
+  expect(screen.getByText("Carbon")).toBeVisible();
+});
+
+test("should render tooltip containing status `timeStamp` when status `text` is hovered", async () => {
+  const user = userEvent.setup();
+  render(
+    <Note
+      createdDate="23 May 2020, 12:08 PM"
+      noteContent={EditorState.createEmpty()}
+      status={{
+        text: "Edited",
+        timeStamp: "23 May 2020, 12:10 PM",
+      }}
+    />
+  );
+
+  const statusText = screen.getByText("Edited");
+  expect(statusText).toBeVisible();
+  await user.hover(statusText);
+
+  const tooltip = await screen.findByText("23 May 2020, 12:10 PM");
+  expect(tooltip).toBeVisible();
+});
+
+test("should render LinkPreviews when passed via the `previews` prop as an array", () => {
+  const previews = [
+    <LinkPreview key="1" url="https://example.com" />,
+    <LinkPreview key="2" url="https://example.com" />,
+  ];
+  render(
+    <Note
+      createdDate="23 May 2020, 12:08 PM"
+      noteContent={EditorState.createEmpty()}
+      previews={previews}
+    />
+  );
+
+  const links = screen.getAllByRole("link");
+
+  expect(links).toHaveLength(2);
+  expect(links[0]).toHaveAttribute("href", "https://example.com");
+});
+
+test("should render LinkPreviews when passed via the `previews` prop as a node", () => {
+  const previews = (
+    <>
+      <LinkPreview key="1" url="https://example.com" />
+      <LinkPreview key="2" url="https://example.com" />
+    </>
+  );
+  render(
+    <Note
+      createdDate="23 May 2020, 12:08 PM"
+      noteContent={EditorState.createEmpty()}
+      previews={previews}
+    />
+  );
+
+  const links = screen.getAllByRole("link");
+
+  expect(links).toHaveLength(2);
+  expect(links[0]).toHaveAttribute("href", "https://example.com");
+});
+
+test("should render with `ActionPopover` when passed via the `inlineControl` prop", async () => {
+  const user = userEvent.setup();
+  const inlineControl = (
+    <ActionPopover>
+      <ActionPopoverItem>Copy</ActionPopoverItem>
+      <ActionPopoverItem>Edit</ActionPopoverItem>
+    </ActionPopover>
+  );
+  render(
+    <Note
+      createdDate="23 May 2020, 12:08 PM"
+      noteContent={EditorState.createEmpty()}
+      inlineControl={inlineControl}
+    />
+  );
+
+  await user.click(screen.getByRole("button", { name: "actions" }));
+
+  expect(screen.getByRole("button", { name: "Copy" })).toBeVisible();
+  expect(screen.getByRole("button", { name: "Edit" })).toBeVisible();
+});
+
+test("should throw when `inlineControls` is not an instance of `ActionPopover`", () => {
+  const spy = jest.spyOn(console, "error").mockImplementation(() => {});
```

Comment:
**praise**: keeping this localised where its needed will make sure we don't accidentially mask any legitimate console errors :+1: